version: '3.7'

services:
  traefik:
    image: "traefik:v2.2.11"
    command:
      - --providers.docker=true
      - --providers.docker.exposedbydefault=false
      - --providers.docker.usebindportip=false
      - --serverstransport.insecureskipverify=true
      - --log.level=DEBUG
      - --accesslog=true
      - --certificatesResolvers.le.acme.storage=/etc/traefik/acme/acme.json
      - --entrypoints.http.address=:80
      - --entrypoints.http.proxyprotocol.insecure
      - --entrypoints.https.address=:443
      - --entrypoints.https.proxyprotocol.insecure
      - --entrypoints.https.http.tls.certResolver=le
      - --api.dashboard
      - --api
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./etc/traefik:/etc/traefik
    ports:
      - ${KITT_IP}:443:443
    labels:
      - "id=traefik"
      - "traefik.enable=true"
      - "traefik.http.routers.traefik-api.entrypoints=https"
      - "traefik.http.routers.traefik-api.rule=HostRegexp(`traefik.{domain:.+}`) && PathPrefix(`/api`)"
      - "traefik.http.routers.traefik-api.service=api@internal"
      - "traefik.http.routers.traefik-dash.entrypoints=https"
      - "traefik.http.routers.traefik-dash.rule=HostRegexp(`traefik.{domain:.+}`)"
      - "traefik.http.routers.traefik-dash.service=dashboard@internal"

volumes:
  traefik:

networks:
  default:
    external:
      name: kitt
